src/__tests__/compileModel.test.ts

```diff
@@ -671,7 +671,7 @@ describe('save', () => {
 
 		const id = 'id';
 		// @ts-expect-error: intentionally invalid data to trigger validation failure
```

Comment:
The issue here was that previously the data was invalid because of the missing prop2. After the type change it was reporting that the ts expect error was unused because prop2 was now correctly optional. I changed it so that the data remained invalid.